Extend information about platforms, requirements and dependencies. (#58)
heremaps/here-olp-edge-sdk-cpp: README.md

Additionally, there are convenient classes for working with the geospatial tiling schemes that are used by most OLP catalog layers.

## Supported platforms

> The SDK has been tested on the following platforms with the following requirements:

Reviewer: ```suggestion
The SDK is actively tested on the following platforms with the following requirements:
```

| Platform | Minimum requirement |
| ------- | -------- |

> | Ubuntu | GCC 5.4 or Clang 7.0 |

Reviewer: ```suggestion
| Ubuntu | GCC 5.4 and Clang 7.0 |
```

Additionally, there are convenient classes for working with the geospatial tiling schemes that are used by most OLP catalog layers.

## Supported platforms
The SDK has been tested on the following platforms with the following requirements:

| Platform | Minimum requirement |
| ------- | -------- |
| Ubuntu | GCC 5.4 and Clang 7.0 |

> | Windows | MSVC++ 2013  |

Reviewer: According to our Azure pipeline setup:
vmImage: 'vs2017-win2016'

| MacOS | Apple Clang 9.1 |
| iOS |  XCode 10.2, Swift 5.0 |
| Android |  API level 21 |

## Dependencies

The dependencies of HERE OLP Edge SDK C++ are the following:

| Library   | Minimum version |
| --------- | --------------- |
| OpenSSL   | 1.0.2 |
| Boost (headers only) | 1.69.0 |
| leveldb   | 1.21 |
| snappy    | 1.1.7 |
| rapidjson | latest |

## Building the SDK

CMake is the main build system. The minimal required version of CMake is 3.5.

### Building on Linux [![Linux build status][1]][2] [![Linux code coverage][5]][6]

> #### Dependencies

Reviewer: Add Curl as a dependency here

Additionally, there are convenient classes for working with the geospatial tiling schemes that are used by most OLP catalog layers.

## Supported platforms
The SDK has been tested on the following platforms with the following requirements:

| Platform | Minimum requirement |
| ------- | -------- |
| Ubuntu | GCC 5.4 and Clang 7.0 |
| Windows | MSVC++ 2017  |
| MacOS | Apple Clang 9.1 |
| iOS |  XCode 10.2, Swift 5.0 |
| Android |  API level 21 |

## Dependencies

The dependencies of HERE OLP Edge SDK C++ are the following:

| Library   | Minimum version |
| --------- | --------------- |
| OpenSSL   | 1.0.2 |
| Boost (headers only) | 1.69.0 |
| leveldb   | 1.21 |
| snappy    | 1.1.7 |
| rapidjson | latest |

## Building the SDK

CMake is the main build system. The minimal required version of CMake is 3.5.

### Building on Linux [![Linux build status][1]][2] [![Linux code coverage][5]][6]

#### Dependencies

> Additionally to the dependencies listed above building the SDK on Linux requires libcurl >=7.47.0.

Reviewer: ```suggestion
Additionally to the dependencies listed above building the SDK on Linux requires libcurl v7.47.0 or later.
```
